motoman_driver/MotoPlus/MotionServer.c

```diff
@@ -1873,3 +1876,25 @@ int Ros_MotionServer_GetDhParameters(Controller* controller, SimpleMsg* replyMsg
 
 	return apiRet;
 }
+
+
+int Ros_MotionServer_SetSelectedTool(Controller* controller, SimpleMsg* receiveMsg, SimpleMsg* replyMsg)
+{
+	int groupNo = receiveMsg->body.selectTool.groupNo;
+	int tool = receiveMsg->body.selectTool.tool;
+
+	if (groupNo >= 0 && groupNo < controller->numRobot)
+	{	
+		if (tool >= 0 && tool < 64)

```

Comment:
Would it be possible to avoid this magic nr? Introduce a constant?

Reply:
Is `64` always the maximum nr of tool configurations? Across all supported controllers?

Reply:
Yes.